src/evo/assetlocktx.h

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2022 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_EVO_ASSETLOCKTX_H
+#define BITCOIN_EVO_ASSETLOCKTX_H
+
+#include <bls/bls_ies.h>
+#include <evo/specialtx.h>
+#include <primitives/transaction.h>
+
+#include <key_io.h>
+#include <serialize.h>
+#include <tinyformat.h>
+#include <univalue.h>
+
+class CBlockIndex;
+class CCreditPool;
+
+class CAssetLockPayload
+{
+public:
+    static constexpr uint16_t CURRENT_VERSION = 1;
+    static constexpr auto SPECIALTX_TYPE = TRANSACTION_ASSET_LOCK;
+
+private:
+    uint16_t nVersion{CURRENT_VERSION};
+    uint16_t nType{0};
+    std::vector<CTxOut> creditOutputs;
+
+public:
+    CAssetLockPayload(uint16_t nType, const std::vector<CTxOut>& creditOutputs) :
+        nType(nType),
+        creditOutputs(creditOutputs)
+    {}
+
+    CAssetLockPayload() = default;
+
+    SERIALIZE_METHODS(CAssetLockPayload, obj)
+    {
+        READWRITE(
+            obj.nVersion,
+            obj.nType,
+            obj.creditOutputs
+        );
+    }
+
+    std::string ToString() const;
+
+    void ToJson(UniValue& obj) const
+    {
+        obj.clear();
+        obj.setObject();
+        obj.pushKV("version", int(nVersion));
+        obj.pushKV("type", int(nType));
+        UniValue outputs;
+        outputs.setArray();
+        for (const CTxOut& out : creditOutputs) {
+            outputs.push_back(out.ToString());
+        }
+        obj.pushKV("creditOutputs", outputs);
+    }
+
+    // getters
+    uint16_t getVersion() const;
+    uint16_t getType() const;
+    const std::vector<CTxOut>& getCreditOutputs() const;
+};
+
+class CAssetUnlockPayload
+{
+public:
+    static constexpr uint16_t CURRENT_VERSION = 1;
+    static constexpr auto SPECIALTX_TYPE = TRANSACTION_ASSET_UNLOCK;
+
+    static constexpr size_t MAXIMUM_WITHDRAWALS = 32;
+
+private:
+    uint16_t nVersion{CURRENT_VERSION};
+    uint64_t index{0};
+    uint32_t fee{0};
+    uint32_t requestedHeight{0};
+    uint256 quorumHash{0};
+    CBLSSignature quorumSig{};
```

Comment:
Maybe you can use `CBLSLazySignature` instead for performance. @UdjinM6 what do you think?